sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -42,15 +42,19 @@ export class DefaultAzureCredential extends ChainedTokenCredential {
   constructor(tokenCredentialOptions?: DefaultAzureCredentialOptions) {
     const credentials = [];
     credentials.push(new EnvironmentCredential(tokenCredentialOptions));
-    credentials.push(new ManagedIdentityCredential(tokenCredentialOptions));
-    if (process.env.AZURE_CLIENT_ID) {
+
+    // In case a user assigned ID has been provided.
+    const managedIdentityClientId =
+      tokenCredentialOptions?.managedIdentityClientId || process.env.AZURE_CLIENT_ID;
+    if (managedIdentityClientId) {
       credentials.push(
-        new ManagedIdentityCredential(
-          tokenCredentialOptions?.managedIdentityClientId || process.env.AZURE_CLIENT_ID,
-          tokenCredentialOptions
-        )
+        new ManagedIdentityCredential(managedIdentityClientId, tokenCredentialOptions)
       );
     }
+
+    // In case the user provided ID doesn't work, but the system assigned ID works.
+    credentials.push(new ManagedIdentityCredential(tokenCredentialOptions));
```

Comment:
Does the ordering of credentials in the chain matters? If so, it does look to me like we flipped the order here - is that on purpose?

Reply:
Before, system managed credential had higher priority than user managed credential. I'll confirm with Scott if this is ok!

Reply:
@schaabs should user managed identities have preference over system managed identities?

Reply:
If a user assigned client id is specified either through the `managedIdentityClientId` option or through the `AZURE_CLIENT_ID` environment variable we should *ONLY* give back token for the user assigned identity. In other words we shoudln't be adding both the credentials to the chain, the secon should only be added in an else clause.

```js    
    if (managedIdentityClientId) {
      credentials.push(
        new ManagedIdentityCredential(managedIdentityClientId, tokenCredentialOptions)
      );
    } else {
       credentials.push(new ManagedIdentityCredential(tokenCredentialOptions));
    }
```